envoy/tracing/trace_context.h

```diff
@@ -93,6 +93,14 @@ class TraceContext {
    * @param val The context value of string view type.
    */
   virtual void setByReference(absl::string_view key, absl::string_view val) PURE;
+
+  /**
+   * Removes the following key and its associated values from the tracing
+   * context.
+   *
+   * @param key The key to remove if it exists.
+   */
+  virtual void removeByKey(absl::string_view key) PURE;
```

Comment:
Look good to me. But should we just name it `remove` or `erase` (like lots of std container)? 

In fact, I always want to simplify these `*ByKey` but have not time and not sure what name would be better for this interface. May be we can start with this method. 🤔

Reply:
I agree that we should rename these at some point e.g. the `*ByKey` removal you suggest.

For consistency with the others I think `ByKey` is reasonable in this change. Remove is present in `HeaderMap` which `RequestHeaderMapImpl` inherits both from `HeaderMap` and `TraceContext`. The params are slightly different which would make it unambigious for the compiler, but might be a bit confusing to readers. Lmk if this is a blocker on your end.

Reply:
You are right. The remove for now may make the readers confused. If we want to do the renaming, it would be better to split the TraceContext from HeaderMap, in the future.